.circleci/config.yml

```
@@ -302,26 +305,72 @@ jobs:
           name: Stop containers
           command: docker-compose -f testing/utp/docker/docker-compose.yml down
 workflows:
-  merge-test:
+  check-build-test:
     jobs:
-      - docker-build
+      - check-workspace-crates
+      - cargo-fmt
+      - cargo-clippy
+      - build
+      - build-windows
+      - test
+      - utp-test
+  docker-master-commit:
+    jobs:
+      - docker-build:
+          name: docker_build_trin
+          target: trin
+          tags: latest
       - docker-publish:
+          name: docker_publish_trin
+          target: trin
           requires:
-            - docker-build
+            - docker_build_trin
           filters:
             branches:
               only: master
-      - docker-build-bridge
-      - docker-publish-bridge:
+      - docker-build:
+          name: docker_build_bridge
+          target: bridge
+          tags: latest
+      - docker-publish:
+          name: docker_publish_bridge
+          target: bridge
           requires:
-            - docker-build-bridge
+            - docker_build_bridge
           filters:
             branches:
               only: master
-      - cargo-fmt
-      - cargo-clippy
-      - build
-      - test
-      - build-windows
-      - utp-test
-      - check-workspace-crates
+  docker-master-tag:
+    when:
+      and: [ << pipeline.git.tag >>, << pipeline.git.branch.is_default >> ]
+    jobs:
+      - docker-build:
+          name: docker_build_trin
+          target: trin
+          tags: << pipeline.git.tag >>-$(git rev-parse --short HEAD) stable prod
+          filters:
+            tags:
+              only: /^v.*$/
```

It might be wise to be a bit stricter, to avoid triggering on a branch like `very-cool-tag`. Maybe something like:
```suggestion
              only: /^v\d+[\d\.]*(-\w*\.\d+)?$/
```
~(I'm not sure what regex dialect circle speaks)~ [They use a Java dialect of regex](https://support.circleci.com/hc/en-us/articles/22413890054939-Regex-for-Branch-Filtering-in-CircleCI).

I tested this against `v12.345.6-alpha.15`

---

Your regex would also match something like: `v123...345`.
I improved it a bit to: `/^v\d+(\.\d+){0,2}(-\w*\.\d+)?$/`